File: Boost/app/src/main/java/com/nowfloats/NavigationDrawer/SiteMeter/SiteMeterModel.java
package com.nowfloats.NavigationDrawer.SiteMeter;

/**
 * Created by guru on 17-06-2015.
 */
public class SiteMeterModel implements Comparable {
    public String Title;
    public String Desc;
    public String Percentage;
    public boolean status;
    public int sortChar;
    public int position;
    public void setStatus(boolean status) {
        this.status = status;
    }
    public void setSortChar(int c){this.sortChar = c;}

    public SiteMeterModel(int position ,String s, String s1, String s2, boolean status,int c) {
        this.position = position;
        this.Title = s;
        this.Desc = s1;
        this.Percentage = s2;
        this.status = status;
        this.sortChar = c;
    }

    @Override
    public int compareTo(Object another) {
        int cVal = ((SiteMeterModel)another).sortChar;
        return (cVal)-this.sortChar;
    }
}

File: dashboard/src/main/java/com/dashboard/model/live/drScore/siteMeter/SiteMeterModel.kt
package com.dashboard.model.live.drScore.siteMeter

import com.dashboard.constant.RecyclerViewItemType
import com.dashboard.recyclerView.AppBaseRecyclerViewItem
import com.dashboard.utils.*
import java.io.Serializable

class SiteMeterModel(
    var position: Int? = null,
    var Title: String? = null,
    var Desc: String? = null,
    var Percentage: String? = null,
    var status: Boolean? = null,
    var sortChar: Int = 0,
    var isPost: Boolean = false,
) : AppBaseRecyclerViewItem, Serializable, Comparable<Any?> {

  var recyclerViewItemType: Int = RecyclerViewItemType.ITEMS_CONTENT_SETUP_ITEM_VIEW.getLayout()

  override fun getViewType(): Int {
    return recyclerViewItemType
  }

  override fun compareTo(other: Any?): Int {
    val cVal = (other as? SiteMeterModel)?.sortChar ?: 0
    return cVal - sortChar
  }


  enum class TypePosition(var value: Int) {
    BUSINESS_NAME(businessName),
    DESCRIPTION(description),
    CATEGORY(category),
    EMAIL(email),
    PHONE(phone),
    ADDRESS(address),
    BUSINESS_HOURS(businessHoursV),
    IMAGE(image),
    LOGO(logo),
    POST(post),
    SOCIAL(social),
    DOMAIN(domain);

    companion object {
      fun fromValue(value: Int?): TypePosition? = values().firstOrNull { it.value == value }
    }
  }
}